lib/arel/table.rb

```diff
@@ -96,6 +96,7 @@ def eql? other
     alias :== :eql?
 
     def type_cast_for_database(attribute_name, value)
+      return value if value.is_a? Arel::Nodes::SqlLiteral
```

Comment:
Are you suggesting?

```ruby
return value unless value.is_a? Arel::Nodes::Node
```

That doesn't work with strings, so it breaks `test/attributes/test_attribute.rb:997`